cmd/metrics.go

```diff
@@ -89,48 +89,40 @@ func (c *minioCollector) Collect(ch chan<- prometheus.Metric) {
 		return
 	}
 
-	storageAPIs := []StorageAPI{}
-	for _, ep := range globalEndpoints {
-		for _, endpoint := range ep.Endpoints {
-			if endpoint.IsLocal {
-				// Construct storageAPIs.
-				sAPI, _ := newStorageAPI(endpoint)
-				storageAPIs = append(storageAPIs, sAPI)
-			}
-		}
-	}
+	storageInfo := objLayer.StorageInfo(context.Background(), true)
 
-	disksInfo, onlineDisks, offlineDisks := getDisksInfo(storageAPIs)
+	offlineDisks := storageInfo.Backend.OfflineDisks
+	onlineDisks := storageInfo.Backend.OnlineDisks
 	totalDisks := offlineDisks.Merge(onlineDisks)
 
-	for _, offDisks := range offlineDisks {
-		// MinIO Offline Disks per node
-		ch <- prometheus.MustNewConstMetric(
-			prometheus.NewDesc(
-				prometheus.BuildFQName("minio", "disks", "offline"),
-				"Total number of offline disks in current MinIO server instance",
-				nil, nil),
-			prometheus.GaugeValue,
-			float64(offDisks),
-		)
-	}
+	// MinIO Offline Disks per node
+	ch <- prometheus.MustNewConstMetric(
+		prometheus.NewDesc(
+			prometheus.BuildFQName("minio", "disks", "offline"),
+			"Total number of offline disks in current MinIO server instance",
+			nil, nil),
+		prometheus.GaugeValue,
+		float64(offlineDisks.Sum()),
+	)
 
-	for _, totDisks := range totalDisks {
-		// MinIO Total Disks per node
-		ch <- prometheus.MustNewConstMetric(
-			prometheus.NewDesc(
-				prometheus.BuildFQName("minio", "disks", "total"),
-				"Total number of disks for current MinIO server instance",
-				nil, nil),
-			prometheus.GaugeValue,
-			float64(totDisks),
-		)
-	}
+	// MinIO Total Disks per node
+	ch <- prometheus.MustNewConstMetric(
+		prometheus.NewDesc(
+			prometheus.BuildFQName("minio", "disks", "total"),
+			"Total number of disks for current MinIO server instance",
+			nil, nil),
+		prometheus.GaugeValue,
+		float64(totalDisks.Sum()),
+	)
 
 	localPeer := GetLocalPeer(globalEndpoints)
-	for _, di := range disksInfo {
+
+	for i := 0; i < len(storageInfo.Total); i++ {
+		mountPoint, total, free := storageInfo.MountPaths[i], storageInfo.Total[i],
+			storageInfo.Available[i]
+
 		// Trim the host
-		absPath := strings.TrimPrefix(di.RelativePath, localPeer)
```

Comment:
Looks like this relative path thingy is needed anymoee from diskInfo struct? Can this be removed

Reply:
We are using it in getStorageInfo() @harshavardhana to calculate the mountPaths

Reply:
Why do we need to keep relativePath? Can we not keep what is needed trimmed and returned always? The relative path approach is a bit unusual as its not directly used anywhere other than trimming.

Reply:
The reason for adding the relativePath is that when we call StorageAPI.DiskInfo().. The call can be a storage REST call for remote disks.. So the posix.DiskInfo() was returning the diskPath alone. So this change will register the Node's address as RelativePath 

```
	if globalIsDistXL {
		localPeer = GetLocalPeer(globalEndpoints)
	}

	return DiskInfo{
		Total:        di.Total,
		Free:         di.Free,
		Used:         used,
		RootDisk:     rootDisk,
		RelativePath: localPeer + s.diskPath,
	}, nil
```

So we need this relative path to know the disk's mountPath which is used in disk-level calculations.

Reply:
For example, a local peer is not a POSIX concept as such @Praveenrajmani we may have done it already but it looks frivolous. 

If you look closely getLocalPeer is not needed at posix.go at all because we do know where we made the call in the first place. For example disk.Hostname() would return localPeer

This type of unexpected association is not necessary, because this information exists already elsewhere. Also, it is hardly RelativePath, it's actually `full path` i.e `peer + path`

Reply:
Agreed. We could use disk.Hostname() for the mountPaths. It would be empty for local disks. We could read the local address in those cases. :+1:

Reply:
I added a commit which removes RelativePath from disk path and replace it with MountPoint